documentation/3_Layers_of_the_Reference_Architecture_Model/3_5_System_Layer/3_5_3_App_Store_and_Data_Apps.md

```diff
@@ -1,43 +1,25 @@
+* INPUT: The input endpoint is considered mandatory for all app types that work with data or data streams. The data input endpoint describes an interface through which data can be transported to an app within the connector's environment.
+* INPUT EXTERNAL: The input external endpoint serves as an interface to connect to external data sources or data streams outside the actual connector environment. This endpoint is particularly relevant for apps of type Adapter App.
+* OUTPUT: The output endpoint is also considered mandatory for app types that transmit data or data streams to other data apps or connectors. The output endpoint describes an interface through which data can be consumed within the connector environment by apps or the connector itself.
+* OUTPUT EXTERNAL: At this endpoint, communication is established beyond the boundaries of the Connector. This special form of output endpoint is primarily relevant for Adapter Apps that establish a connection to an external data source. Reading data from external data sources is made possible by this endpoint. 
+
+Further endpoints, besides the ones listed above, are the config endpoint and the status endpoint. The config endpoint can be used to actively set or change configuration parameters during the runtime of an IDS Data App. Optional is the so-called status endpoint, which can be used to retrieve status information from an IDS Data App during runtime.
+
+The IDS App Store is a secure platform for distributing Data Apps. An IDS App Store consists of a registry for available Data Apps in this App Store. It also features the capabilty to search for Data Apps using different search options (e.g. by functional or non-functional properties, pricing model, certification status, community ratings, etc.). Therefore, an App Store must support operations for Data App registration, publication, maintenance, and query, as well as operations for the provisioning of a Data App to a connector to App Users as depicted in Figure [3.5.3.1](#AppStoreArchitecture). These basic operations can be optionally complemented by additional services, e.g. billing or support activities.
+
+![AppStoreArchitecture](./media/app-store-architecture.png)
+#### _Fig. 3.5.3.1: App Store Architecture_
+
+An IDS App Store also consists of an IDS Connector in order to communicate with the Connectors of App Providers and App Users within the Data Space. As a consequence, each instance of an App Store must be compliant to the Connector Certification criteria an provide the functionalities and endpoints of general Connectors together with the above-mentioned operations (e.g. provide a Self-Description, have a valid IDS Identity and use a valid DAT in its communication).
```

Comment:
Please consider also the case that a developer is able to upload Data Apps without using an IDS Connector.
Also, the certification process foresees a two-step verification process for software artefacts (such as Data Apps):
* Every data app is automatically checked by a test suite before being published in the App Store. 
* Optionally, for some software artefacts there is manual evaluation procedure for software artefacts.

So could you please add this:
* A developer (depicted by a user icon) is able to upload Data Apps to the App Store
* The App Registration and Publication Service triggers automatic checks before publishing (depicted as a link to an external test service).

Reply:
@gbrost I have modified the architecture figure to reflect the changes. I have a couple of doubts:

- What component is this automatic testing and verification process part of? Is it part of the App Store itself, the Certification Body or something else? It looks a little odd as an independent service.
- Please check the [Data App Publication process](https://github.com/International-Data-Spaces-Association/IDS-RAM_4_0/blob/data-apps/documentation/3_Layers_of_the_Reference_Architecture_Model/3_3_Process_Layer/3_3_5_Publishing_and_using_Data_Apps.md) figure to see if the process makes sense. From my understading, uploading the App Image is always done without the connector, then the publishing of the data app can be done through a connector or without it. Is that correct?

I will modify the text in both files to reflect the changes once these doubts are clarified.